### ISSUE TITLE
Nested Lorri projects are endlessly evaluated

### ISSUE THREAD

**Issue by evanjs** [target/lorri#268]
**Describe the bug**

Unsure if a bug per-se, but lorri seems to go crazy when lorri-enabled projects are nested

**To Reproduce**
1. `lorri init`
2. `mkdir a && cd a`
3. `lorri init`
4. `cd ..`

**Expected behavior**

Run `watch -n1 "ls /nix/var/nix/gcroots/auto | wc -l"` and watch the auto gcroots go up like crazy
It seems like Lorri considers other Lorri environments when evaluating changes

**Metadata**

```console
$ lorri info
lorri version: 1
Lorri Project Configuration

expression: /home/evanjs/src/<project_path>/shell.nix
```

```console
$ uname -a
Linux sekka 5.4.7 #1-NixOS SMP Tue Dec 31 15:46:36 UTC 2019 x86_64 GNU/Linux
```

**Additional context**

Not sure if this is a valid/supported use-case, but maybe we should handle nested Lorri projects differently?
i.e. not evaluate child projects or something.  Not sure what is most practical.
___
#51 seems sort of similar; I wonder if nested Lorri projects would be a good candidate for a default ignore in Lorri

**Comment by evanjs** [comment]
Update: This seems to still happen after disabling lorri in my project's child directories.
Not sure why this is happening.  I thought it might've been due to `src` being set to a child directory of the project, but the problem persists after setting `src` to `fetchGit` pointing to the project's remote.

**Comment by curiousleo** [comment]
Thanks for the clear repro steps! I can repro this with the lorri version on `nixos-19.09`, but not with lorri built from `master`.

The logs on `master` show a bunch of "ignoring spurious metadata change event within the profiles dir". This was added in https://github.com/target/lorri/pull/260 to fix #176.

It seems likely that you've found a way to trigger #176.

**Comment by curiousleo** [comment]
We should probably cut a release soon that includes this fix. Related: https://github.com/target/lorri/issues/269
